### DISCUSSION TITLE
Question about connection establishment time

### DISCUSSION THREAD

**Discussion by farmboy-dev** [oracle/python-oracledb#334] [Q&A]
Hello python-oracledb team,
I've been seeing some abnormal connection establishment time. it usually takes less than 1s to establish connection to OracleDB, but sometime some connections take longer (5s~60s) than usual. 
Could you guide me how to debug this issue? 
Thank you!

-- connection info -- 
python-oracledb 2.2.0
code 
```
dsn = 
(DESCRIPTION =
        (ADDRESS = (PROTOCOL = TCP)(HOST = server1-vip)(PORT = port))
        (ADDRESS = (PROTOCOL = TCP)(HOST = server2-vip)(PORT = port))
        (FAILOVER=ON)(LOAD_BALANCE=OFF)
        (CONNECT_DATA=
        (SERVICE_NAME = NAME)
        )
        (SERVER=dedicated)
        (FAILOVER_MODE= (TYPE=SESSION)(METHOD=BASIC)(RETRIES=20)(DELAY=5)
        (CONNECT_TIMEOUT=1)(RETRY_COUNT=3)
        )
s = time.time()
oracledb.init_oracle_client()
connection = oracledb.connect(user="USER", password="PWD", dsn=dsn)
e = time.time()
print(f"Connection established: {e-s}")
```

**Comment by sreguna** [comment]
There seems to be a missing close parenthesis for FAILOVER_MODE. 
(FAILOVER_MODE= (TYPE=SESSION)(METHOD=BASIC)(RETRIES=20)(DELAY=5)**)**
The CONNECT_TIMEOUT and RETRY_COUNT parameters should be at the DESCRIPTION level. Can you please try with the modified connect string. With this each connect attempt should failover after 1 sec and if both sites are down you should see a timeout error within a few seconds.

**Reply by farmboy-dev** [reply]
Thank you for the comment ; )
I edited it, but there are still some connections that take longer than usual.

**Comment by farmboy-dev** [comment]
it was a rootless docker's network issue. After changing the network driver, it's working as expected.
